src/clean-core/span.hh

```diff
@@ -31,7 +31,7 @@ public:
 
     /// generic span constructor from contiguous_range
     /// CAUTION: container MUST outlive the span!
-    template <class Container, cc::enable_if<is_contiguous_range<Container, T>> = true>
+    template <class Container, cc::enable_if<is_contiguous_range<Container, T> && !std::is_same_v<std::decay_t<Container>, span>> = true>
```

Comment:
why do you need the additional check? the default generated copy/move ctors are not hidden by this and the code is also correct for `span const&&`.

Reply:
Without this check, this ctor is used for span-to-span copies. It's correct but very annoying when stepping into a function that takes multiple spans as arguments.